src/Repository/Notification.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Friendica\Repository;
+
+use Exception;
+use Friendica\BaseRepository;
+use Friendica\Core\Hook;
+use Friendica\Model;
+use Friendica\Collection;
+use Friendica\Network\HTTPException\NotFoundException;
+use Friendica\Util\DateTimeFormat;
+
+class Notification extends BaseRepository
```

Comment:
I was wondering about naming the repository, module and collection the same way than the base table?

Reply:
Yes, I'm not sure if it's good to have 5 Notification classes in different namespaces .. Any ideas how to prevent it ? I'd say adding the Postfix of the class-type will increase readability:
- `Factory\NotificatonFactory`
- `Repository\NotificationRepository 
- `Model\Notification` (here I'd say we let it as is, because that's the "real" notification item in the codebase
- `Object\NotificationObject`

Or we alter the view-only namespace, like:
- `Factory\NotificationDTOFactory`
- `Object\NotificationDTO` (DTO = Data Transfer Object)

Reply:
> Yes, I'm not sure if it's good to have 5 Notification classes in different namespaces .. Any ideas how to prevent it ?

Yes, by naming the repository, module and collection the same way than the related database table.

It would give:
- `Repository\Notify`
- `Model\Notify`
- `Collection\Notifies`
- `Factory\Notification`
- `Factory\Introduction`
- `Object\Notification` and possibly `Collection\Notifications`
- `Object\Introduction` and possibly `Collection\Introductions`

Reply:
Hm .. In a FollowUp, I would have renamed the table from `notify` to `notification` to make it more clear.. So we would be in the same situation like before after such a change ^^

Reply:
Ha ha, I've tried to suggest field/table name changes in the past but @annando has always discouraged me, so it just is a fact of life I've accepted.

Reply:
I see ^^ .. So I'd say I rename it as suggested (what about the "DTO" postfix to mark objects as "transfer only" objects to distinguish them from "real" model instances?)

Reply:
I don't like the suffix at all (even less than `notify` as a noun!). I don't think the underlying design pattern should show in the class name.

Reply:
I could have thought that ^^ .. We'll see if it's possible for all classes this way